### DISCUSSION TITLE
How to print relative file path, current file name and line number in nelua

### DISCUSSION THREAD

**Discussion by ghost** [edubart/nelua-lang#136] [Q&A] [answered]
Example :
```lua
-- main.nelua
print(relative file path, file name, line number)
-- the print statement is at line number 15
```
the output should look something like this
```
src/main.nelua:15:
```

**Comment by edubart** [comment, marked as answer]
This is what you wanted:

```lua
print(#[srcloc.srcname..':'..srcloc.lineno]#) -- outputs "test.nelua:1"
```

The `srcloc` variable is a preprocessor table that always have the source location for the current node being processed.

You could also create a polymorphic function that gets instantiated at every call, and print the source location where the polymorphic function was instantiated using `polysrcloc`:

```lua
local function f() <alwayspoly>
  print(#[polysrcloc.srcname..':'..polysrcloc.lineno]#)
end
f() -- outputs "test.nelua:5"
```

The `polysrcloc` variable is a preprocessor table that always have the source location where the current polymorphic function being processed was instantiated.

Both ways is supported since commit https://github.com/edubart/nelua-lang/commit/2fe098d91780aaf8a95fca9daeb831bf4b9a987d

**Reply by ghost** [reply]
As always, Thanks :)
